ndcube/tests/test_ndcube.py

```diff
@@ -1119,6 +1119,13 @@ def test_cube_arithmetic_multiply_notimplementederror(ndcube_2d_ln_lt_units):
         _ = ndcube_2d_ln_lt_units * ndcube_2d_ln_lt_units
 
 
+@pytest.mark.parametrize('power', [2, -2, 10, 0.5])
+def test_cube_arithmetic_power(ndcube_2d_ln_lt_uncert, power):
+    cube_quantity = u.Quantity(ndcube_2d_ln_lt_uncert.data, ndcube_2d_ln_lt_uncert.unit)
+    new_cube = ndcube_2d_ln_lt_uncert ** power
+    check_arithmetic_value_and_units(new_cube, cube_quantity**power)
+
+
```

Comment:
We also need tests for when for the following cases:
* `self.uncertainty is None`
* `self.unit is None`

These can probably be done in the same test.